shared/src/commonMain/kotlin/dev/johnoreilly/confetti/di/Koin.kt

```diff
@@ -29,14 +29,103 @@ fun commonModule() = module {
     single { AppSettings(get()) }
 }
 
+@OptIn(ApolloExperimental::class)
 fun createApolloClient(sqlNormalizedCacheFactory: NormalizedCacheFactory): ApolloClient {
     val memoryFirstThenSqlCacheFactory = MemoryCacheFactory(10 * 1024 * 1024)
         .chain(sqlNormalizedCacheFactory)
 
     return ApolloClient.Builder()
         .serverUrl("https://confetti-349319.uw.r.appspot.com/graphql")
-        //.serverUrl("http://10.0.2.2:8080/graphql")
+//        .serverUrl("http://10.0.2.2:8080/graphql")
         .addHttpInterceptor(LoggingInterceptor())
-        .normalizedCache(memoryFirstThenSqlCacheFactory, writeToCacheAsynchronously = true)
+        .normalizedCache(
+            normalizedCacheFactory = memoryFirstThenSqlCacheFactory,
+            cacheKeyGenerator = TypePolicyCacheKeyGenerator,
+            metadataGenerator = CursorPaginationMetadataGenerator(setOf("SessionConnection")),
+            apolloResolver = FieldPolicyApolloResolver,
+            recordMerger = FieldRecordMerger(CursorPaginationFieldMerger()),
+            writeToCacheAsynchronously = true
+        )
         .build()
 }
+
+@OptIn(ApolloExperimental::class)
+@Suppress("UNCHECKED_CAST")
+private class CursorPaginationMetadataGenerator(private val connectionTypes: Set<String>) :
```

Comment:
This (and `CursorPaginationFieldMerger`) would probably reside in the library, as it should work the same in all Relay-based pagination APIs. For now I copied/pasted it from [the test in the library](https://github.com/apollographql/apollo-kotlin/pull/4237/files?w=1#diff-6b5012b1b3780a2db4c73cfef61a99c727e36cab2197c5e5f426bf01fa0e0545R314).